File: problems/1266-minimum-time-visiting-all-points/solution.c
int minTimeToVisitAllPoints(int** points, int pointsSize, int* pointsColSize) {
    int totalTime = 0;
    for (int i = 0; i < pointsSize - 1; i++) {
        int x1 = points[i][0];
        int y1 = points[i][1];
        int x2 = points[i + 1][0];
        int y2 = points[i + 1][1];
        int dx = abs(x2 - x1);
        int dy = abs(y2 - y1);
        totalTime += (dx > dy) ? dx : dy;
    }
    return totalTime;
}

File: problems/1266-minimum-time-visiting-all-points/solution.py
class Solution:
    def minTimeToVisitAllPoints(self, points: list[list[int]]) -> int:
        time = 0
        for i in range(len(points) - 1):
            x1, y1 = points[i]
            x2, y2 = points[i + 1]
            dx = abs(x2 - x1)
            dy = abs(y2 - y1)
            time += max(dx, dy)
        return time